tests/jobwatcher/unittests.py

```diff
@@ -7,32 +7,34 @@
 
 class OptimalNodeCountTests(unittest.TestCase):
```

Comment:
we should test get_optimal_nodes with gpus as well if not done already (you could move this tests into the pytest format)

and also you might want to take a look at the test coverage to understand if you missed any critical code path

Reply:
Testing `get_required_nodes` which includes `get_optimal_nodes` in `test_get_required_nodes_mix` and `test_get_required_nodes_gpu`